0000-product.md

```diff
@@ -0,0 +1,463 @@
+- Feature Name: Product
+- Start Date: 5/13/19
+- RFC PR: [#5](https://github.com/hyperledger/grid-rfcs/pull/5)
+
+# Summary
+[summary]: #summary
+
+
+This RFC proposes a generic and extensible framework for a Hyperledger Grid
+Product entity as well as a more specific - and fully encapsulated - GS1
+compliant Product.
+
+GS1 is a widely used data standard in enterprises and, given that positioning
+and familiarity, Grid support feels natural.  Additional implementations of
+Product for specialized industries or use cases may derive from or extend this
+implementation.
+
+_For the base implementation of a product on grid, there will be 4 fields. A
+product_id, a product_namespace, an owner, and properties (a repeated key-value
+field)._
+
+
+# Motivation
+[motivation]: #motivation
+
+The Grid Product implementation is designed for sharing product master data
+between participants. Product is a near universal concept within supply chain
+solutions and would naturally be one of the highest areas of re-use across Grid
+applications.
+
+The desgn will address use cases including:
+
+- Sharing of Product master data across a network
+- Including Product in other business transactions (track and trace events,
+  purchases, sales, etc)
+- Enriching UX experiences by including additional attribution of a Product such
+  as names or descriptions
+
+It is also useful to use product information as auxiliary data in other supply
+chain solutions.  For example, in track and trace, the location and temperature
+of an item (an instance of a product) is stored.  When presenting this
+information later to a human user, it is useful to combine it with information
+about the product, such as the product’s name and description.  Therefore,
+Product becomes a reusable and important component within Grid.
+
+# Guide-Level Explanation
+[guide-level-explanation]: #guide-level-explanation
+
+## Entities
+
+A **__product__** is an archetype of an item that is transacted, traded, or
+referenced in a supply chain.  Each product has a **__product_namespace__**.
+This RFC defines a single product_namespace for GS1; a product in the GS1
+product_namespace is called a GS1 product.  Note that this design supports
+extending additional product namespaces in the future.
+
+A product is referenced using a **product_id**.  For GS1 products, the
+product_id is a Global Trade Item Number (GTIN), which is part of the GS1
+specifications.
+
+Each product has an **__owner__**, which is the identifier of the organization
```

Comment:
Have we thought about transfer of ownership scenarios and how that might be handled?

Reply:
Transfer of ownership in the sense an organization will create a product and transfer ownership to another organization? I think that would introduce a change in the GTIN for that product, since the gtin is derived from the organization_id? 

I don't think this transfer scenario is accounted for and could be addressed at a later time after some discussion around it. I can capture the lack of this functionality in the "Drawbacks" section

Reply:
Pepsi owns Aquafina - Co Prefix 123456 (Pepsi is owner, Aquafina is Brand)
Pepsi sells Aquafina to  Beverage Q. Beverage Q is now the owner, Aquafina is the brand. Company prefix belongs to Beverage Q

Reply:
Gotcha, I think we would need to discuss this use case to fully implement it. The concept of transferring ownership is not included in this RFC, and can be included at a later time. I will make note of this in the drawbacks section.

Reply:
@genasmorgan How does GS1 manage authoritative ownership of co prefixes today? Is there some 'root of trust' which is published? How do you ensure an organization has the authority to publish a GTIN under a prefix, for example.

Reply:
The short answer is that various Member Organizations issue company prefixes to organizations US based companies receive company prefixes from GS1 US. Member organizations have various mechanisms for verifying a company prefix was issued to an organization. Additionally,  a global service is being developed "Verified by GS1" that will provide a verification that a GTIN (and seven attributes) was in fact commissioned by the owning organization.

Reply:
It would be very good if there was a cryptographic (signature-scheme) based way to publish and verify these claims. For example, maybe a gs1 key signs the initial creation and transfer of company prefixes to the owning organization, and then that owning organization's key signs the creation of subordinate GTINs and co prefix transfer events.

Reply:
Yes, we are working on that as a future extension to Verified by GS1. The Verified by GS1 service is just launching this fall (two countries on board with six more, including the US, to be online by the end of the year). We are starting with data - cryptographic signatures/verifiable claims are being piloted by GS1 with various players in the W3C decentralized identifiers/community credentials groups.